framework/grid/DataColumn.php

```diff
@@ -189,6 +189,12 @@ protected function renderFilterCellContent()
             if (is_array($this->filter)) {
                 $options = array_merge(['prompt' => ''], $this->filterInputOptions);
                 return Html::activeDropDownList($model, $this->attribute, $this->filter, $options) . $error;
+            } elseif ($this->format === 'boolean') {
+                $options = array_merge(['prompt' => ''], $this->filterInputOptions);
+                return Html::activeDropDownList($model, $this->attribute, [
+                        $this->grid->formatter->booleanFormat[0],
+                        $this->grid->formatter->booleanFormat[1]
+                    ], $options) . $error;
```

Comment:
I suggest this way of writing (align closing square bracket with the beginning of "return" and reduce indentation accordingly):

```php
return Html::activeDropDownList($model, $this->attribute, [
    $this->grid->formatter->booleanFormat[0],
    $this->grid->formatter->booleanFormat[1]
], $options) . $error;
```